src/sources/splunk_hec.rs

```diff
@@ -445,6 +447,33 @@ impl<R: Read> Stream for EventStream<R> {
     }
 }
 
+/// Parse a `i64` unix timestamp that can either be in seconds, milliseconds or
+/// nanoseconds.
+///
+/// This will first attempt tp arse millaseconds and check that the date
+/// produced is larger than the year 2001. This is due to the fact that we can
+/// successfully parse milliseconds via the seconds fn and get incorrect dates.
+/// To enforce that we get correct dates and don't incorrectly parse a
+/// millisecond timestamp we check this date. If that fails we then attempt to
+/// parse it as seconds then nano seconds.
+fn parse_timestamp(t: i64) -> DateTime<Utc> {
+    if let LocalResult::Single(t) = Utc.timestamp_millis_opt(t) {
+        if t.date().year() > 2001 {
```

Comment:
Ok so this one is weird, if you try to parse nano seconds from the milliseconds fn you get `LocalResult::None`. Because I can't get the playground to update here is an example https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=e8f4d597ccce1175e49b96d74b892809 this panics where in our code it returns the variants.

So this states that if we are nano seconds both sec and milli sec will fail. The problem here is parsing seconds as milli seconds which can produce a valid date but that is much younger. So here we are assuming all logs are created after 2001 and therefore if we somehow produce a date that is < 2001 then we know its probably milliseconds. Otherwise, there is no way to detect if the incoming value is seconds or milliseconds. We could choose a better date but then again I don't see why 2001 is an issue either. 

(was supposed to post this earlier)